src/solver.rs

```diff
@@ -0,0 +1,41 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+use ::pool::Pool;
+use ::queue::Queue;
+use libsolv_sys::Solver as _Solver;
+use libc;
+
+pub struct Solver {
+    pub(crate) ctx: Rc<RefCell<Pool>>,
+    pub(crate) _s: *mut _Solver,
+}
+
+impl Solver {
+    pub(crate) fn new_with_context(ctx: Rc<RefCell<Pool>>) -> Self {
+        use libsolv_sys::solver_create;
+        let _s = {
+            let borrow = ctx.borrow_mut();
+            unsafe {solver_create(borrow._p)}
+        };
+        Solver{ctx: ctx, _s: _s}
+    }
+
+    pub(crate) fn new_with_solver(ctx: Rc<RefCell<Pool>>, _s: *mut _Solver) -> Self {
+        Solver{ctx: ctx, _s: _s}
+    }
+
+    pub fn solve(&mut self, job: &mut Queue) -> libc::c_int {
```

Comment:
+    pub fn solve(&mut self, job: &mut Queue) -> libc::c_int {
ignatenkobrain
it should rather return vec of Problem's
ignatenkobrain
yeah, it's mostly wrapping stuff, but would be awesome to do that :D
abaxter
describe what you'd like it to do in detail. I'm up for making changes
abaxter
I mean, there's not much there
ignatenkobrain
so, solve() returns number of problems which you can analyze later out of Solv* object.
ignatenkobrain
in all bindings, it returns list of Porblem objects
ignatenkobrain
typedef struct {                         Solver *solv;                         Id id;                       } Problem;
ignatenkobrain
https://github.com/openSUSE/libsolv/blob/master/bindings/solv.i#L3045
ignatenkobrain
so basically it creates Problem object which has back reference to Solver and problem ID
ignatenkobrain
and there are some solv_*() methods which work with those problems
ignatenkobrain
I will be back in  few hours, taking a small bike trip
abaxter
k
ignatenkobrain
abaxter: we will need to discuss features in ext part
ignatenkobrain
once I will be back
ignatenkobrain
abaxter: https://github.com/fedora-modularity/depchase/blob/master/depchase#L363-L368